### PR TITLE
feat : added watchlater

### PR DESCRIPTION
**Pull request by JuvitaSaini09**
Added Watch later feature 
Preview Link- https://deploy-preview-9--dreamy-stroopwafel-3d0999.netlify.app/

### CODE REVIEW

**Review by muskaanshah — APPROVED** [review]
Functionality working as expected.
Suggestion: UI changes - 
1. cursor point when hovering on create playlist modal's close button
2. When create playlist modal opens, the side section of single video page shifts, which is bad UX
3. Try using custom scrollbar or remove scrollbar in the side section of single video page
4. Give the drawer a fixed height. Every time you click on new tab, the height of the drawer changes
5. Scrollbar appearing on create playlist modal even when it's not required

File: src/components/playlistModal/playlistModal.js [hunk 1]
```diff
@@ -2,15 +2,17 @@ import React, { useState } from "react";
...
   deleteVideoFromPlaylistApi,
 } from "../../util/playlistApiCall";
+import { addToWatchLater,  removeFromWatchLater } from "../../util/watchLaterApi";
 import { Toast } from "../allComponents";
 import "./playlistModal.css";
-
 function PlaylistModal() {
+  const {videosInPlaylist,setwatchLaterCheckbox,setWatchLaterCheckboxDisptach}=useWatchLaterCheckbox(); 
   const {setToast,toastState,toastDispatch}=useToast();
   const {playlistNameValue,setPlaylistNameValue}=usePlaylistVideoContext();
```
Comment (Adarsh-Patel-Dev) [inline src/components/playlistModal/playlistModal.js:17]: you can destructure properties of line 17, 18 in one line as they are coming from ` usePlaylistVideoContext`

File: src/server.js [hunk 2]
```diff
@@ -88,6 +100,14 @@ export function makeServer({ environment = "development" } = {}) {
       this.post("/user/likes", addItemToLikedVideos.bind(this));
       this.delete("/user/likes/:videoId", removeItemFromLikedVideos.bind(this));
 
+      // watch later routes (private)
+      this.get("/user/watchlater", getWatchLaterVideosHandler.bind(this));
+      this.post("/user/watchlater", addItemToWatchLaterVideos.bind(this));
```
Comment (gauravsinhaweb) [inline src/server.js:105]: good use of `bind`

File: src/pages/watchLater/watchLater.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      <Sidebar />
+      <div className="col2">
+        {videosInPlaylist[0] !== undefined && <h1 className="playlistHeading">WATCH LATER VIDEOS</h1>}
+        {/* <---------History Videos-------------> */}
+        <div className="videosListing mt-3">
+          {videosInPlaylist === undefined || videosInPlaylist[0] === undefined ? <EmptyPage text="Empty Watch Later" /> :<>
+          {videosInPlaylist.map((item) => {
+            return (
+              <div className="video-card" key={item._id}>
+                <Link onClick={() => setSingleVideo(item)} to="/video">
```
Comment (gauravsinhaweb) [inline src/pages/watchLater/watchLater.js:34]: you can destructure item here then use it.

**Review by gauravsinhaweb — APPROVED** [review]
Good to merge!

File: src/context/watchLaterContext/watchLaterContext.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  )
+}
+
+const useWatchLaterCheckbox=()=>useContext(watchLaterCheckboxContext);
+
+export  {WatchLaterProvider , useWatchLaterCheckbox}
+
+/* ---------------------------------------------------------
+   ---------------------------------------------------------
+//const {watchLaterCheckbox,setwatchLaterCheckbox,setWatchLaterCheckboxDisptach,videosInPlaylist}=useWatchLaterCheckbox();
```
Comment (Adarsh-Patel-Dev) [inline src/context/watchLaterContext/watchLaterContext.js:48]: remove the comments

**Review by Adarsh-Patel-Dev — APPROVED** [review]
- Fix the height of Aside bar on all pages except homepage also fix the position of Asidebar so it don't scroll with page
- Playlist functionality is not completely working (adding videos in playlist is not working)

Suggestions on UI:
- Reduce the height of Navbar
-increase width of video title
-reduce the size of delete icon

**Review by prankurpandeyy — APPROVED** [review]

**Comment by prankurpandeyy** [comment]
code looks good to me expect the side bar is breaking

**Pull request merged**